### DISCUSSION TITLE
How do I generate GraphQL query from SQL

### DISCUSSION THREAD

**Discussion by digitalsimboja** [supabase/supabase#6297] [Questions] [answered]
I am trying to find a way to see how this was generated from the Supabase Dashboard. Could anyone guide me how to automatically generate a GraphQL query when I run the SQL query to create the tables. Is this possible?

**Comment by olirice** [comment, marked as answer]
Hi @digitalsimboja

> automatically generate a GraphQL query when I run the SQL query to create the tables

Im not sure exactly what you mean here. If you're trying to figure out how to generate a valid GraphQL query, you may want to check out [this discussion](https://github.com/supabase/supabase/discussions/6144) that shows how to connect GraphiQL, a GraphQL IDE, that will let you explore the schema + lint and execute queries in the browser.

In case I misunderstood, and you were asking how to query via GraphQL in SQL, the following snippet shows how to do that as well

```sql
-- Create a table
create table book(
	id serial primary key,
	title text not null
);

-- Populate a record
insert into book(title)
select 'A Guide to GraphQL at Supabase';

-- Rebuild the graphql schema from the sql schema
select graphql.rebuild_schema();

-- Execute a GraphQL query
select graphql_public.graphql(
	"operationName":= null,
	query := $$
    {
        bookCollection(first: 1) {
            edges {
                 node {
                     id
                     title
                 }
            }
        }
    }
	$$,
	variables := null
);
```
returns
```json
{
  "data": {
    "bookCollection": {
      "edges": [
        {
          "node": {
            "id": 1,
            "title": "A Guide to GraphQL at Supabase"
          }
        }
      ]
    }
  }
}
```

**Comment by digitalsimboja** [comment]
Thanks @olirice for quick response. Okay what I meant was how to generate the Schema from the database table? 
If this does generate the schema, where can I find the schema being generated: 

-- Rebuild the graphql schema from the sql schema

`select graphql.rebuild_schema();`

**Reply by digitalsimboja** [reply]
Okay I got it [here](https://github.com/supabase-community/supabase-graphql-example)

Thanks and that solves it
